src/vs/platform/userDataProfile/electron-main/userDataProfileStorageIpc.ts

```diff
@@ -66,7 +67,7 @@ export class ProfileStorageChangesListenerChannel extends Disposable implements
 		keys = keys.filter(key => key !== TARGET_KEY);
 		if (keys.length) {
 			const keyTargets = loadKeyTargets(this.storageMainService.applicationStorage.storage);
-			profileStorageValueChanges.push({ profile: this.userDataProfilesService.defaultProfile, changes: keys.map(key => ({ key, scope: StorageScope.PROFILE, target: keyTargets[key] })) });
+			profileStorageValueChanges.push({ profile: this.userDataProfilesService.defaultProfile, changes: keys.map(key => ({ key, scope: StorageScope.PROFILE, target: keyTargets[key], source: StorageChangeSource.SELF })) });
```

Comment:
Do we really know the source is `SELF`? Should we maybe just leave it out here, since it seems to be optional?

Reply:
Or if it cannot be optional, are we certain `SELF` is correct here? I am not sure we can even distinguish at this level?

Reply:
Oh, I think Sandeep was suggesting to remove the `source` if possible.